python/lsst/ap/pipe/metrics.py

```diff
@@ -139,3 +140,57 @@ def run(self, matchedFakes, band):
             self.log.info("Nothing to do: no matched catalog found.")
             meas = None
         return Struct(measurement=meas)
+
+
+class ApFakesCountMetricConnections(
+        ApFakesCompletenessMetricConnections,
+        dimensions={"instrument", "visit", "detector", "band"},
+        defaultTemplates={"coaddName": "deep",
+                          "fakesType": "fakes_",
+                          "package": "ap_pipe",
+                          "metric": "apFakesCompleteness"}):
+    pass
+
+
+class ApFakesCountMetricConfig(
+        ApFakesCompletenessMetricConfig,
+        pipelineConnections=ApFakesCountMetricConnections):
+    """ApFakesCompleteness config.
+    """
+    pass
+
+
+class ApFakesCountMetricTask(ApFakesCompletenessMetricTask):
+    """Metric task for summarizing the completeness of fakes inserted into the
+    AP pipeline.
+    """
+    _DefaultName = "apFakesCount"
+    ConfigClass = ApFakesCountMetricConfig
+
+    def run(self, matchedFakes, band):
+        """Compute the completeness of recovered fakes within a magnitude
+        range.
+
+        Parameters
+        ----------
+        matchedFakes : `lsst.afw.table.SourceCatalog` or `None`
+            Catalog of fakes that were inserted into the ccdExposure matched
+            to their detected counterparts.
+
+        Returns
+        -------
+        result : `lsst.pipe.base.Struct`
+            A `~lsst.pipe.base.Struct` containing the following component:
+            ``measurement``
+                the ratio (`lsst.verify.Measurement` or `None`)
+        """
+        if matchedFakes is not None:
+            magnitudes = matchedFakes[f"{self.config.magVar}" % band]
+            magCutFakes = matchedFakes[np.logical_and(magnitudes > self.config.magMin,
+                                                      magnitudes < self.config.magMax)]
+            meas = Measurement(self.config.metricName,
+                               len(magCutFakes) * u.count)
+        else:
+            self.log.info("Nothing to do: no matched catalog found.")
```

Comment:
Do you expect this task to be run on data without a matched catalog? If it is not expected that `matchedFakes` would be `None`, then this should at least be `self.log.warn`.

Reply:
I might suggest slightly changing the wording, as well, to reflect that this task does not try to load a catalog itself: "Nothing to do: no matched catalog supplied."

Reply:
I'm okay with this being info. It's consistent with the other metrics we current have.